lib/routes.js

```diff
@@ -2,70 +2,73 @@
 
 var request = require('./request');
 
-module.exports = function(server, socket) {
+module.exports = function(server, socket, socketNamespace) {
   var routingTable = server.table();
 
   routingTable.forEach(function(connection) {
     var routes = connection.table.filter(function(item) {
       return item.settings &&
-             item.settings.plugins &&
-             item.settings.plugins['hapi-io'];
+          item.settings.plugins &&
+          item.settings.plugins['hapi-io'];
     });
 
     routes.forEach(function(route) {
       var hapiio = route.settings.plugins['hapi-io'];
       var event = typeof hapiio === 'string' ? hapiio : hapiio.event;
+      var namespace = (!(typeof hapiio === 'string') && !(typeof hapiio.namespace === 'string')) ? '/' : hapiio.namespace;
```

Comment:
There's no need to check if`hapiio.namespace` is a string.

We check if `hapiio` is a string because we allow the user to use an event name or an object, when configuring a route:

```js
plugins: {
  'hapi-io': 'get-admin-user'
}
```

```js
plugins: {
  'hapi-io': {
    'event': 'get-admin-user',
    'namespace': '/admin'
  }
}
```